packages/expo-updates/android/src/main/java/expo/modules/updates/statemachine/UpdatesStateMachine.kt

```diff
@@ -117,6 +117,7 @@ class UpdatesStateMachine(
       UpdatesStateEventType.CheckCompleteUnavailable to UpdatesStateValue.Idle,
       UpdatesStateEventType.CheckError to UpdatesStateValue.Idle,
       UpdatesStateEventType.Download to UpdatesStateValue.Downloading,
+      UpdatesStateEventType.DownloadProgress to UpdatesStateValue.Downloading,
```

Comment:
was thinking that whether we need a DownloadProgress state? can we add progress to Downloading?

if the transition is like `Downloading -> DownloadProgress -> Downloading -> DownloadProgress -> ...` then it feels too much.
also how about this case can possibly happen? `DownloadProgress -> DownloadComplete` 

cc @douglowder for the comment from the update state machine expert

Reply:
Agreed... @intergalacticspacehighway is not adding a new state here, but a new event and new context property. I think this is fine!